alluxio/alluxio_file_system.py

```diff
@@ -674,3 +674,510 @@ def _validate_path(self, path):
             raise ValueError(
                 "path must be a full path with a protocol (e.g., 'protocol://path')"
             )
+
+
+class AlluxioAsyncFileSystem:
+    """
+    Access Alluxio file system
+
+    Examples
+    --------
+    >>> # Launch Alluxio with ETCD as service discovery
+    >>> alluxio = AlluxioFileSystem(etcd_host=localhost)
+    >>> # Or launch Alluxio with user provided worker list
+    >>> alluxio = AlluxioFileSystem(worker_hosts="host1,host2,host3")
+
+    >>> print(alluxio.listdir("s3://mybucket/mypath/dir"))
+    [
+        {
+            "mType": "file",
+            "mName": "myfile",
+            "mLength": 77542
+        }
+
+    ]
+    >>> print(alluxio.read("s3://mybucket/mypath/dir/myfile"))
+    my_file_content
+    """
+
+    ALLUXIO_PAGE_SIZE_KEY = "alluxio.worker.page.store.page.size"
+    ALLUXIO_PAGE_SIZE_DEFAULT_VALUE = "1MB"
+    ALLUXIO_SUCCESS_IDENTIFIER = "success"
+    LIST_URL_FORMAT = "http://{worker_host}:{http_port}/v1/files"
+    FULL_PAGE_URL_FORMAT = (
+        "http://{worker_host}:{http_port}/v1/file/{path_id}/page/{page_index}"
+    )
+    PAGE_URL_FORMAT = "http://{worker_host}:{http_port}/v1/file/{path_id}/page/{page_index}?offset={page_offset}&length={page_length}"
+    WRITE_PAGE_URL_FORMAT = (
+        "http://{worker_host}:{http_port}/v1/file/{path_id}/page/{page_index}"
+    )
+    GET_FILE_STATUS_URL_FORMAT = "http://{worker_host}:{http_port}/v1/info"
+    LOAD_SUBMIT_URL_FORMAT = (
+        "http://{worker_host}:{http_port}/v1/load?path={path}&opType=submit"
+    )
+    LOAD_PROGRESS_URL_FORMAT = (
+        "http://{worker_host}:{http_port}/v1/load?path={path}&opType=progress"
+    )
+    LOAD_STOP_URL_FORMAT = (
+        "http://{worker_host}:{http_port}/v1/load?path={path}&opType=stop"
+    )
```

Comment:
I assume these configurations are the same as those in the `AlluxioFileSystem` class? Would be better to move them to a separate class to remove duplication.

Reply:
move to class global name works?